pilot/pkg/networking/util/util.go

```diff
@@ -490,10 +490,6 @@ func BuildLbEndpointMetadata(network network.ID, tlsMode, workloadname, namespac
 		FilterMetadata: map[string]*pstruct.Struct{},
 	}
 
-	if network != "" {
```

Comment:
We use this in `epNetwork := network.ID(istioMetadata(lbEp, "network"))`

Reply:
Ack, if it's just for internal usage then I bet we can pass it along in a more optimal way. Will try it out. The tests are failing so looks like we have test coverage of this

Reply:
I am also using it in https://github.com/istio/istio/pull/33047 to implement network aware